enos/enos.py

```diff
@@ -306,33 +309,70 @@ def init_os(env=None, **kwargs):
 
 
 @enostask("""
-usage: enos bench [--scenarios=SCENARIOS] [--times=TIMES]
-                  [--concurrency=CONCURRENCY] [--wait=WAIT]
+usage: enos bench [--workload=WORKLOAD]
                   [-vv|-s|--silent]
 
 Run rally on this OpenStack.
 
 Options:
   -h --help                 Show this help message.
-  --scenarios=SCENARIOS     Name of the files containing the scenarios
-                            to launch. The file must reside under the
-                            rally directory.
-  --times=TIMES             Number of times to run each scenario
-                            [default: 1].
-  --concurrency=CONCURRENCY Concurrency level of the tasks in each
-                            scenario [default: 1].
-  --wait=WAIT               Seconds to wait between two scenarios
-                            [default: 0].
+  --workload=WORKLOAD       Path to the workload directory.
+                            This directory must contain a run.yml file
+                            that contains the description of the different
+                            scenarios to launch
 """)
 def bench(env=None, **kwargs):
+    def cartesian(d):
+        """returns the cartesian product of the args."""
+        logging.debug(d)
+        f = []
+        for k, v in d.items():
+            if isinstance(v, list):
+              f.extend([[[k, vv] for vv in v]])
+            else:
+              f.append([[k,v]])
+        logging.debug(f)
+        product = []
+        for e in itertools.product(*f):
+            product.append(dict(e))
+        return product
+
     logging.debug('phase[bench]: args=%s' % kwargs)
-    playbook_path = os.path.join(ANSIBLE_DIR, 'run-bench.yml')
+    workload_dir = kwargs["--workload"]
+    with open(os.path.join(workload_dir, "run.yml")) as workload_f:
+        workload = yaml.load(workload_f)
+        for nature, desc in workload.items():
+            scenarios = desc.get("scenarios", [])
+            for scenario in scenarios:
+                # merging args 
+                top_args = desc.get("args", {})
+                args = scenario.get("args", {})
+                top_args.update(args)
+                # merging enabled, skipping if not enabled
+                top_enabled = desc.get("enabled", True)
+                enabled = scenario.get("enabled", True)
```

Comment:
Do we wanna implicitly enable scenario, in this case giving the `True` default value is OK, or do we wanna explicitly enable scenario, in this case we should put `False` as the default value?